### ISSUE TITLE
Socket Exception - javapns-jdk16-168.jar

### ISSUE THREAD

**Issue by mwkirk** [mwkirk/javapns#57]
_Original author: dux.dee...@gmail.com (July 20, 2011 13:26:18)_

Following is the error I get on my windows machine :

Connection initialized...
preclose
in finally
java.net.SocketException: Socket is closed
    at com.sun.net.ssl.internal.ssl.SSLSocketImpl.checkEOF(Unknown Source)
    at com.sun.net.ssl.internal.ssl.AppInputStream.read(Unknown Source)
    at sun.nio.cs.StreamDecoder.readBytes(Unknown Source)
    at sun.nio.cs.StreamDecoder.implRead(Unknown Source)
    at sun.nio.cs.StreamDecoder.read(Unknown Source)
    at java.io.InputStreamReader.read(Unknown Source)
    at java.io.BufferedReader.fill(Unknown Source)
    at java.io.BufferedReader.readLine(Unknown Source)
    at java.io.BufferedReader.readLine(Unknown Source)
    at javapns.back.PushNotificationManager.sendNotification(Unknown Source)
    at Push.main(Push.java:62)

I am using the sample example provided :

 import javapns.back.PushNotificationManager;
 import javapns.back.SSLConnectionHelper;
 import javapns.data.Device;
 import javapns.data.PayLoad;

public class Push {

```
 private static final String HOST = &quot;gateway.sandbox.push.apple.com&quot;;
 private static final int PORT = 2195;
 private static final int BADGE = 66;

 private static String iPhoneId = &quot;my device id&quot;;
 private static String certificate = &quot;APNS_SSLCertificate_Key.p12&quot;;      //Generating this as per http://code.google.com/p/javapns/wiki/GetAPNSCertificate

 private static String passwd = &quot;1289&quot;;


public static void main( String[] args ) throws Exception {
   try {
         PayLoad aPayload = new PayLoad();
         aPayload.addBadge( BADGE );

         PushNotificationManager pushManager = PushNotificationManager.getInstance();


         pushManager.addDevice(&quot;iPhone&quot;, iPhoneId);

         Device client = pushManager.getDevice( &quot;iPhone&quot; );
         System.out.println( &quot;Client setup successfull.&quot; );

         pushManager.initializeConnection( HOST, PORT, certificate, passwd, SSLConnectionHelper.KEYSTORE_TYPE_PKCS12);
         System.out.println( &quot;Connection initialized...&quot; );


        // Send message
         pushManager.sendNotification( client, aPayload );
         System.out.println( &quot;Message sent!&quot; );


        System.out.println( &quot;# of attempts: &quot; + pushManager.getRetryAttempts() );
         pushManager.stopConnection();


        System.out.println( &quot;done&quot; );


    } catch (Exception e) {
         e.printStackTrace();
     }
 }
```

 }

_Original issue: http://code.google.com/p/javapns/issues/detail?id=57_

**Comment by mwkirk** [comment]
_From idbill.p...@gmail.com on July 25, 2011 21:35:37_
Seems like a bad cert.

What does the SSL debugging show?

**Comment by mwkirk** [comment]
_From dux.dee...@gmail.com on July 28, 2011 10:27:58_
Yup...it was a bad cert.
Problem solved, but my code is stuck up after printing "Connection initialized..."
Whatmay be the error ?

Thanks

**Comment by mwkirk** [comment]
_From idbill.p...@gmail.com on August 30, 2011 19:59:04_
try the next version

**Comment by mwkirk** [comment]
_From sype...@gmail.com on September 07, 2011 16:29:45_
Fixed in 2.0 (see issue #54).
